packages/cozy-stack-client/src/FileCollection.js

```diff
@@ -891,15 +900,24 @@ class FileCollection extends DocumentCollection {
    * async createDirectoryByPath - Creates one or more folders until the given path exists
    *
    * @param  {string} path - Path of the created directory
+   * @param {DirectoryAttributes} [attributes] - Attributes of the created directory
    * @returns {object} The document corresponding to the last segment of the path
    */
-  async createDirectoryByPath(path) {
+  async createDirectoryByPath(path, attributes) {
     const parts = path.split('/').filter(part => part !== '')
     const root = await this.statById(ROOT_DIR_ID)
     if (!parts.length) return root
     let parentDir = root
-    for (const part of parts) {
-      parentDir = await this.getDirectoryOrCreate(part, parentDir.data)
+    for (const [i, part] of parts.entries()) {
+      if (i === parts.length - 1) {
+        parentDir = await this.getDirectoryOrCreate(
+          part,
+          parentDir.data,
+          attributes
+        )
+      } else {
+        parentDir = await this.getDirectoryOrCreate(part, parentDir.data)
+      }
```

Comment:
Can you explain why you have to make this check and this different call for this one?

Reply:
It is because we want to add attributes only to the final directory (the "target" directory). 

I updated the js doc to be more explicit.

Reply:
The explanation seems complicated to me, so I find the API hard to understand. I wonder if we need a method that does both at the same time, why not simply have: 
- a method that creates the folder 
- a method that puts the metadata you want on this folder 

Or a method "set attributes on the root of this path" or something like that?

Reply:
**Summary of discussions and work** 

/!\ It is not possible to update the metadata of a folder and it is not possible to add metadata after folder creation. 

So if we want to create a folder _foo/bar/backup_ with attributes on _backup_
- we have a method to create _/foo/bar_ (createDirectoryByPath)
- we have a method to create _backup_ inside _/foo/bar_ with the folder attributes (createDirectory)
=> result is easy to understand and there is not a lot of code inside cozy-client

Implementation differences https://github.com/cozy/cozy-flagship-app/compare/a0b184e5c9d125496e9006026be43702e2a697b8..5c05e7a07a9fc265384ee2d18b2e8a61f40bc6e3

Reply:
I know that this is not optimal for now, maybe we should accept to update the metadata of a folder even after its creation?